tee-worker/litentry/core/assertion-build/src/dynamic/scripts/deploy.ts

```diff
@@ -0,0 +1,322 @@
+ */
+function encryptBuffer(pubKey: crypto.KeyLike, plaintext: Uint8Array): Buffer {
+    const bs = 384 // 3072 bits = 384 bytes
+    const bsPlain = bs - (2 * 256) / 8 - 2 // Maximum plaintext block size
+    const count = Math.ceil(plaintext.length / bsPlain) // Use Math.ceil to ensure proper chunk count
+
+    const cipherText = Buffer.alloc(bs * count)
+
+    for (let i = 0; i < count; i++) {
+        const plainSlice = plaintext.slice(
+            i * bsPlain,
+            Math.min((i + 1) * bsPlain, plaintext.length)
+        )
+        const cipherSlice = crypto.publicEncrypt(
+            {
+                key: pubKey,
+                padding: crypto.constants.RSA_PKCS1_OAEP_PADDING,
+                oaepHash: 'sha256',
+            },
+            plainSlice
+        )
+
+        cipherSlice.copy(cipherText, i * bs)
+    }
+
+    return cipherText
+}
+
+async function main() {
+    const chain = process.env.CHAIN as string
+    const contract = process.env.CONTRACT as string
+    const secrets = (process.env.SECRETS as string)
+        .split(' ')
+        .filter((secret) => !!secret)
+
+    if (!SUPPORTED_CHAINS.includes(chain)) {
+        throw new Error(
+            `Unsupported chain ${chain}, need be one of ${SUPPORTED_CHAINS}.`
+        )
+    }
+
+    const chainId = `${chainIdPrefix}${chain}`
+    const { rpcs, enclaveRpcs } = getChain(chainId)
+
+    const api = await ApiPromise.create({
+        provider: new WsProvider(rpcs[0].url),
+        types,
+    })
+
+    const encryptedSecrets: string[] = []
+    if (secrets.length > 0) {
+        let teeShieldingKey: KeyObject
+        try {
+            teeShieldingKey = await retrieveTeeShieldingKey(
+                api,
+                enclaveRpcs[0].url
+            )
+        } catch (e) {
+            throw new Error(`Fail to retrieve teeShieldingKey, error: ${e}`)
+        }
+
+        secrets.forEach((secret) => {
+            const encodedSecret = api.createType('String', secret).toU8a()
+            // Some secrets are too large, so need using segment encryption.
+            const encryptedSecret = encryptBuffer(
+                teeShieldingKey,
+                encodedSecret
+            )
+            encryptedSecrets.push(`0x${encryptedSecret.toString('hex')}`)
+        })
+    }
+
+    const ContractFactory = await ethers.getContractFactory(contract)
+    const contractId = generateContractId(ContractFactory.bytecode, secrets)
+
+    console.log(
+        `Begin to deploying contract: ${contract}, to chain: ${chain}, contract id: ${contractId}`
+    )
+
+    const waitForDeployPromise = waitForDeploy(api)
+
+    const proposal = api.tx.evmAssertions.createAssertion(
+        contractId,
+        ContractFactory.bytecode,
+        encryptedSecrets
+    )
+    const alice = genesisSubstrateWallet('Alice')
+    await api.tx.developerCommittee
+        .execute(proposal, proposal.encodedLength)
+        .signAndSend(alice)
```

Comment:
I don't think it will work on prod?

Not for rococo/litmus/litentry either

Reply:
added an overriding option to use wallet to deploy: https://github.com/litentry/litentry-parachain/pull/2948/commits/3ab4c426ff3986f6ff013291b209f531654411c8